components/properties/src/script.rs

```diff
@@ -564,3 +567,78 @@ impl<'data> ScriptWithExtensions<'data> {
+    Result<DataPayload<ScriptWithExtensionsPropertyV1Marker>, PropertiesError>;
+
+/// Returns a [`ScriptWithExtensions`] struct that represents the data for the Script
+/// and Script_Extensions properties.
+///
+/// # Examples
+///
+/// ```
+/// use icu_properties::Script;
+/// use icu::properties::{script, script::ScriptWithExtensions};
+/// use zerovec::ZeroVec;
+///
+/// let provider = icu_testdata::get_provider();
+/// 
+/// let payload =
+///     script::get_script_extensions(&provider)
+///         .expect("The data should be valid");
+/// let data_struct = payload.get();
+/// let swe: &ScriptWithExtensions = &data_struct.data;
+/// 
+/// // get the `Script` property value
+/// assert_eq!(swe.get_script_val(0x0640), Script::Common); // U+0640 ARABIC TATWEEL
+/// assert_eq!(swe.get_script_val(0x0650), Script::Inherited); // U+0650 ARABIC KASRA
+/// assert_eq!(swe.get_script_val(0x0660), Script::Arabic); // // U+0660 ARABIC-INDIC DIGIT ZERO
+/// assert_eq!(swe.get_script_val(0xFDF2), Script::Arabic); // U+FDF2 ARABIC LIGATURE ALLAH ISOLATED FORM
+/// 
+/// // get the `Script_Extensions` property value
+/// assert_eq!(
+///     swe.get_script_extensions_val(0x0640) // U+0640 ARABIC TATWEEL
+///         .iter().collect::<Vec<Script>>(),
+///     vec![Script::Arabic, Script::Syriac, Script::Mandaic, Script::Manichaean, Script::PsalterPahlavi, Script::Adlam, Script::HanifiRohingya, Script::Sogdian, Script::OldUyghur]
+/// );
+/// assert_eq!(
+///     swe.get_script_extensions_val('🥳' as u32) // U+1F973 FACE WITH PARTY HORN AND PARTY HAT
+///         .iter().collect::<Vec<Script>>(),
+///     vec![Script::Common]
+/// );
+/// assert_eq!(
+///     swe.get_script_extensions_val(0x200D) // ZERO WIDTH JOINER
+///         .iter().collect::<Vec<Script>>(),
+///     vec![Script::Inherited]
+/// );
+/// assert_eq!(
+///     swe.get_script_extensions_val('௫' as u32) // U+0BEB TAMIL DIGIT FIVE
+///         .iter().collect::<Vec<Script>>(),
+///     vec![Script::Tamil, Script::Grantha]
+/// );
+/// 
+/// // check containment of a `Script` value in the `Script_Extensions` value
+/// // U+0650 ARABIC KASRA
+/// assert!(!swe.has_script(0x0650, Script::Inherited)); // main Script value
+/// assert!(swe.has_script(0x0650, Script::Arabic));
+/// assert!(swe.has_script(0x0650, Script::Syriac));
+/// assert!(!swe.has_script(0x0650, Script::Thaana));
+/// 
+/// // get a `UnicodeSet` for when `Script` value is contained in `Script_Extensions` value
+/// let syriac = swe.get_script_extensions_set(Script::Syriac);
+/// assert!(syriac.contains_u32(0x0650)); // ARABIC KASRA
+/// assert!(!syriac.contains_u32(0x0660)); // ARABIC-INDIC DIGIT ZERO
+/// assert!(!syriac.contains_u32(0xFDF2)); // ARABIC LIGATURE ALLAH ISOLATED FORM
+/// assert!(syriac.contains_u32(0x0700)); // SYRIAC END OF PARAGRAPH
+/// assert!(syriac.contains_u32(0x074A)); // SYRIAC BARREKH
+/// ```
+pub fn get_script_extensions<D>(provider: &D) -> ScriptWithExtensionsResult
```

Comment:
Note: The other way to do this would be

```
ScriptWithExtensions::try_new(&provider)
```

But I think what you did here is better because it's more consistent with how other properties work.

Reply:
Yep, consistency is the most important criterion here.

And FWIW, all things be equal, I would prefer the global static function version rather than a static function tied to a struct, even though they probably are equivalent, just because it doesn't bias the user against considering [partial application functions](https://functional.works-hub.com/learn/functional-programming-jargon-in-rust-1b555#partial-application) / currying, if they truly want them.  Although maybe this doesn't make a perf difference in Rust, and partial functions probably aren't big in a lot of other ICU4X target programming languages.